lib/babel/transformation/transformers/es6/parameters.rest.js

```diff
@@ -5,6 +5,64 @@ var t    = require("../../../types");
 
 exports.check = t.isRestElement;
 
+var memberExpressionVisitor = {
+  enter: function (node, parent, scope, state) {
+    var localDeclar = scope.getBindingIdentifier(state.name);
+    if (localDeclar !== state.outerDeclar) return;
+
+    if (t.isFunctionDeclaration(node) || t.isFunctionExpression(node)) {
+      state.isOptimizable = false;
+      this.skip();
+      return;
+    }
+
+    if (!t.isReferencedIdentifier(node, parent, { name: state.name })) return;
+
+    if (parent.type === 'MemberExpression') {
```

Comment:
Best to use the `t.isX` helpers instead of directly checking the `type` property for consistency.